torchbenchmark/models/timm_efficientnet/config.py

```diff
@@ -0,0 +1,48 @@
+# Generated by gen_timm_models.py
+import torch
+import torch.nn as nn
+import dataclasses
+from timm.optim import create_optimizer
+
+def resolve_precision(precision: str):
+    assert precision in ('amp', 'float16', 'bfloat16', 'float32')
+    use_amp = False
+    model_dtype = torch.float32
+    data_dtype = torch.float32
+    if precision == 'amp':
+        use_amp = True
+    elif precision == 'float16':
+        model_dtype = torch.float16
+        data_dtype = torch.float16
+    elif precision == 'bfloat16':
+        model_dtype = torch.bfloat16
+        data_dtype = torch.bfloat16
+    return use_amp, model_dtype, data_dtype
+
+@dataclasses.dataclass
+class OptimizerOption:
+    lr: float
+    opt: str
+    weight_decay: float
+    momentum: float
+
+class TimmConfig:
+    def _init_input(self):
+        self.example_inputs = torch.randn(
+            (self.batch_size,) + self.input_size, device=self.device, dtype=self.data_dtype)
+
+    def __init__(self, model, device, precision):
+        self.model = model
+        self.device = device
+        self.use_amp, self.model_dtype, self.data_dtype = resolve_precision(precision)
+        # Configurations
+        self.batch_size = 2
```

Comment:
why use bs=2? i'm not familiar with the particular models, but it's generally important to pick batch sizes that make sense given the real use case.  

it's also common to use different batch sizes for training vs inference, and possibly different based on cpu/gpu.   we haven't done a good job across the suite with this.  

i wonder if you could implement a different batch size for train vs eval, based on what is common for that model in the original repo?

Reply:
The original repo uses batch_size=256 for both train and inference: https://github.com/rwightman/pytorch-image-models/blob/master/benchmark.py#L64. I think it would be too large for torchbench. Is there any numbers I can use for reference, e.g. in other models?

Reply:
I think it would be reasonable to choose the largest (power of 2) batch size you could fit on the GPU, and do a batch size of 1 for inference.

Reply:
> I think it would be reasonable to choose the largest (power of 2) batch size you could fit on the GPU, and do a batch size of 1 for inference.

Sorry but I am not very familiar with ML. How to estimate the largest batch size to fit on the GPU? The CI machine uses a Telsa T4 GPU with 16GB DDR6 memory.

Reply:
haha i don't think there is much to it,  just try 256, 128, ... until it doesn't give OOM error :)

Reply:
> haha i don't think there is much to it, just try 256, 128, ... until it doesn't give OOM error :)

I see. On the CI machine the highest available batch_size without OOM error is 64, so I am using 64 for train and 1 for inference.